File: java_solutions_1-10/Euler_7.java
/*
/   Project Euler problem #7 10001 prime
/
/   What is the 10001 prime number?
/
*/

import static java.lang.Math.sqrt;


public class Euler_7
{
    public static boolean isPrime(int n)
    {
        if (n < 2) return false;
        else if (n == 2) return true;

        for (int i = 3; i < sqrt(n) + 1; i++)
            if (n % i == 0) return false;

        return true;
    }

    public static void main(String[] args)
    {
        int i = 3;
        for (int primeCount=1; primeCount < 10001; i+=2)
            if (isPrime(i)) primeCount++;

        System.out.println("Answer: " + (i - 2));
    }
}

File: Python/py_solutions_1-10/Euler_7.py
# What is the 10001st prime number?

from __future__ import print_function

import math
import timeit

try:
    range = xrange
except NameError:
    pass

start = timeit.default_timer()

def is_prime(n):
    if n == 2:
        return True
    if n == 1 or n % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(n) + 1), 2):
        if n % i == 0:
            return False
    return True

def prime_gen():
    number_prime = n = 0
    while number_prime < 10001:
        if is_prime(n):
            number_prime += 1
            yield n
        n += 1

def euler_7():
    return [i for i in prime_gen()][-1]

print("Answer: {}".format(euler_7()))
stop = timeit.default_timer()
print("Time: {0:9.5f}".format(stop - start))    
